### DISCUSSION TITLE
possible packet_type bug in packet.py

### DISCUSSION THREAD

**Discussion by mattbanhamtektowr** [miguelgrinberg/python-engineio#234] [Q&A]
was just trying to get fxcm.py to connect via socket_io via engineio. Whilst debugging, I think I've found a bug in packet.py. The code reads:
.......
            else:
                self.packet_type = int(encoded_packet[0])
              
However, the valid socket_io message I got back from the fxcm server was 96:0{........}
In other words the OPEN packet type isn't the first character of encoded_packet but the character after the colon.

**Comment by miguelgrinberg** [comment]
What version of the Socket.IO protocol does your server use? The `96:` prefix feels a lot like the old encoding in the v3 version of the engine.io protocol. It's been removed in v4, so the current version of this package does not do that anymore. See https://github.com/socketio/engine.io-protocol#difference-between-v3-and-v4.
